File: android/assets/scripts/gate-door.py
from com.mirage.model.scripts import IScript
from com.mirage.model import Model
from com.mirage.controller import Controller
from com.mirage.model.scripts import ScriptUtils
from com.badlogic.gdx.maps import MapProperties

class Script(IScript):
    def run(self, properties):
        player = Model.INSTANCE.getPlayer()
        obj = properties.get("object")
        if player == obj:
            door = Model.INSTANCE.findObject("gate-door")
            if 10 < float(player.properties.get("x")) < 20 and 10 < float(player.properties.get("y")) < 16:
                if door.properties.get("state") == "closed":
                    door.properties.put("state", "opening")
                    door.properties.put("animation", "MAIN_GATE_OPEN")
                    Controller.INSTANCE.getScreen().addObjectDrawer(door)
                    props = MapProperties()
                    props.put("state", "opened")
                    ScriptUtils.INSTANCE.launchScriptAfterDelay("gate-state-update.py", 1000L, props)
            else:
                if door.properties.get("state") == "opened":
                    door.properties.put("state", "closing")
                    door.properties.put("animation", "MAIN_GATE_CLOSE")
                    Controller.INSTANCE.getScreen().addObjectDrawer(door)
                    props = MapProperties()
                    props.put("state", "closed")
                    ScriptUtils.INSTANCE.launchScriptAfterDelay("gate-state-update.py", 1000L, props)

File: android/assets/scripts/gate-door.lua
args = ...

u = args["actions"]

local objProps = args["object"]:getProperties()
if objProps:get("type") == "entity" then
        local x = objProps:get("x")
        local y = objProps:get("y")
        local door = u:findObject("gate-door")
        local doorprops = door:getProperties()
        if ((x > 10) and (x < 20) and (y > 10) and (y < 16)) then
            if doorprops:get("state") == "closed" then
                doorprops:put("state", "opening")
                doorprops:put("animation", "MAIN_GATE_OPEN")
                u:updateObjectDrawer(door)
                local props = {["state"] = "opened"}
                u:runScriptAfterDelay("gate-state-update", props, 1000)
            end
        elseif doorprops:get("state") == "opened" then
            doorprops:put("state", "closing")
            doorprops:put("animation", "MAIN_GATE_CLOSE")
            u:updateObjectDrawer(door)
            local props = {["state"] = "closed"}
            u:runScriptAfterDelay("gate-state-update", props, 1000)
        end
end